views/FanView/FanView.tsx

```diff
@@ -56,15 +61,60 @@ export const FanView: FC<FanViewProps> = ({ route }) => {
     sendAPIRequest({
       ...route.params,
       action: 'status',
-      additionalParams: 'V_DIRECTION',
+      additionalParams: 'V_PERCENTAGE',
     });
     sendAPIRequest({
       ...route.params,
       action: 'status',
-      additionalParams: 'V_PERCENTAGE',
+      additionalParams: 'V_DIRECTION',
     });
   };
 
+  const checkInput = (
+    value: string,
+    inputChange: (value: string) => void,
+    min: number,
+    max: number,
+  ) => {
+    if (value == '') {
+      value = min.toString();
+    }
+    const potentialValue = parseFloat(value);
+    if (isNaN(potentialValue)) {
+      return;
+    }
+    const potentialValueString = potentialValue.toString();
+    if (
+      value == potentialValueString + ',' ||
+      value == potentialValueString + '.,'
+    ) {
+      value = potentialValueString + '.';
+    }
+    let numberOfDots = value.match(numberRegex)?.length;
+    if (numberOfDots && numberOfDots > 1) {
+      value = potentialValueString + '.';
+      numberOfDots = value.match(numberRegex)?.length;
+      if (numberOfDots && numberOfDots > 1) {
+        value = potentialValueString;
+      }
+    }
+    const finalValueFloat = parseFloat(value);
+    if (finalValueFloat < min) {
+      inputChange(min.toString());
+    } else if (finalValueFloat > max) {
+      inputChange(max.toString());
+    } else {
+      numberOfDots = value.match(numberRegex)?.length;
+      const finalValueString =
+        value == finalValueFloat.toString()
+          ? finalValueFloat.toString()
+          : numberOfDots && numberOfDots > 0
+          ? finalValueFloat.toString() + '.'
+          : finalValueFloat.toString();
+      inputChange(finalValueString);
+    }
```

Comment:
This code is really complex and hard to read. You can simplify it like this
```
    const newValue = value.replace(',', '.').split('.').slice(0, 2).join('.');
    const newValueNumeric = parseFloat(newValue);
    if (newValueNumeric < min) {
      inputChange(min.toString());
    } else if (newValueNumeric > max) {
      inputChange(max.toString());
    } else {
      inputChange(newValue);
    }
```
Maybe you can also use `Math.min` and `Math.max` to simplify this `if` statements, but I've got a problem with a comma. But you can give it a try :D